modules/page-memory/src/test/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/CheckpointProgressImplTest.java

```diff
@@ -64,19 +66,22 @@ void testReason() {
 
     @Test
     void testNextCheckpointNanos() {
-        assertThat(new CheckpointProgressImpl(0).nextCheckpointNanos(), lessThanOrEqualTo(nanoTime()));
-
-        assertThat(new CheckpointProgressImpl(-1).nextCheckpointNanos(), lessThanOrEqualTo(nanoTime()));
-
-        assertThat(new CheckpointProgressImpl(Long.MIN_VALUE).nextCheckpointNanos(), lessThanOrEqualTo(nanoTime()));
+        assertThat(
+                new CheckpointProgressImpl(0).nextCheckpointNanos(),
+                allOf(greaterThan(0L), lessThanOrEqualTo(nanoTime()))
```

Comment:
I just realized that these assertions are wrong! Nano time can be negative, there's nothing wrong with it. I'm not even sure about proper assertions. When it comes to time, things become very tricky.
You can do the following, for example:
- long before = nanoTime();
- long cp = new CheckpointProgressImpl(0).nextCheckpointNanos();
- long after = nanoTime();
- assert cp - before >= 0;
- assert after - cp >= 0;

Assertions for "year" will be different, obviously.

Reply:
Tried to fix it.